docs/adr/adr-004-state-interaction.md

```diff
@@ -57,6 +57,19 @@ type StateAccessor interface {
 3. **LOCAL**: eventually, **full** nodes will be able to provide state to the network, and can therefore execute and 
 respond to the queries without relaying queries to celestia-core (*to be scoped out and implemented in later iterations*)
 
+### Verification of Balances
+In order to check that the balances returned via the `AccountBalance` query are correct, it is necessary to also request
+Merkle proofs from the celestia-app and verify them against the latest head's `AppHash`. In order for the `StateAccessor`
```

Comment:
What if a new block arrives in between requesting a proof and receiving it? Shouldn't the node verify the proof against _a given_ block? Should clarify this case.

Reply:
Good point. I think we need to look deeper into the API which requests proof. Maybe it provides a block height/hash so we can verify proofs specifically against the identified header. 

cc @renaynay

Reply:
@adlerjohn, after looking into this again I think your case is an implementation detail that can be resolved once the above is implemented. I think we can merge without this clarification